packages/moti/src/core/motify.tsx

```diff
@@ -15,45 +17,44 @@ import Animated, {
 import type { MotiProps } from './types'
 import { useMotify } from './use-motify'
 
+export interface MotifyComponent<
+  Props extends object,
+  Ref,
+  Animate = ViewStyle | ImageStyle | TextStyle
+> extends ForwardRefExoticComponent<
+    Props & AnimatedProps<Props> & MotiProps<Animate> & RefAttributes<Ref>
+  > {}
+
 export default function motify<
   Props extends object,
   Ref,
   Animate = ViewStyle | ImageStyle | TextStyle
->(ComponentWithoutAnimation: ComponentType<Props>) {
-  const Component = Animated.createAnimatedComponent(
-    ComponentWithoutAnimation as FunctionComponent<Props>
+>(
+  BaseComponent: ComponentType<Props>
+): () => MotifyComponent<Props, Ref, Animate> {
+  const AnimatedComponent = Animated.createAnimatedComponent(
+    BaseComponent as FunctionComponent<any>
   )
 
   const withAnimations = () => {
```

Comment:
I'm a bit confused why we don't just create the component immediately, instead of returning a function.